packages/app-desktop/gui/NoteEditor/utils/contextMenu.ts

```diff
@@ -53,17 +55,52 @@ function handleCopyToClipboard(options: ContextMenuOptions) {
 	}
 }
 
-export function menuItems(): ContextMenuItems {
+export async function openItemById(options: any, dispatch: Function) {
+	const itemId = options.resourceId || options.itemId;
+	const item = await BaseItem.loadItemById(itemId);
+
+	if (!item) throw new Error(`No item with ID ${itemId}`);
+
+	if (item.type_ === BaseModel.TYPE_RESOURCE) {
+		const localState = await Resource.localState(item);
```

Comment:
Declare the resource here, with type checking: `const resource = item as ResourceEntity` and use that resource variable for this block of code.

Reply:
Done